Samples/MediaPlayerCPP/MainPage.xaml.cpp

```diff
@@ -212,47 +212,57 @@ void MainPage::ExtractFrame(Platform::Object^ sender, Windows::UI::Xaml::RoutedE
-            bool exactSeek = grabFrameExactSeek->IsOn;
             // extract frame using FFmpegInterop and current position
-            auto frameGrabber = co_await FrameGrabber::CreateFromStreamAsync(stream);
-            auto frame = co_await frameGrabber->ExtractVideoFrameAsync(mediaPlayer->PlaybackSession->Position, exactSeek);
-            auto filePicker = ref new FileSavePicker();
-            filePicker->SuggestedStartLocation = PickerLocationId::VideosLibrary;
-            filePicker->DefaultFileExtension = ".jpg";
-            filePicker->FileTypeChoices->Insert("Jpeg file", ref new Platform::Collections::Vector<String^>(1, ".jpg"));
+            frameGrabber = co_await FrameGrabber::CreateFromStreamAsync(stream);
+        }
+        else if (uri != nullptr)
+        {
+            frameGrabber = co_await FrameGrabber::CreateFromUriAsync(uri);
+        }
+        else
+        {
+            DisplayErrorMessage("Please open a video file first.");
+            co_return;
+        }
+        auto frame = co_await frameGrabber->ExtractVideoFrameAsync(mediaPlayer->PlaybackSession->Position, exactSeek);
+        auto filePicker = ref new FileSavePicker();
+        filePicker->SuggestedStartLocation = PickerLocationId::VideosLibrary;
+        filePicker->DefaultFileExtension = ".jpg";
+        filePicker->FileTypeChoices->Insert("Jpeg file", ref new Platform::Collections::Vector<String^>(1, ".jpg"));
 
-            // Show file picker so user can select a file
-            auto file = co_await filePicker->PickSaveFileAsync();
-            if (file != nullptr)
+        // Show file picker so user can select a file
+        auto file = co_await filePicker->PickSaveFileAsync();
+        if (file != nullptr)
+        {
+            auto stream = co_await file->OpenAsync(FileAccessMode::ReadWrite);
+            // encode frame as jpeg file
+            co_await frame->EncodeAsJpegAsync(stream);
+            stream = nullptr;
+
+            // launch file after creation
+            auto launched = co_await Windows::System::Launcher::LaunchFileAsync(file);
+            if (!launched)
             {
-                auto stream = co_await file->OpenAsync(FileAccessMode::ReadWrite);
-                // encode frame as jpeg file
-                co_await frame->EncodeAsJpegAsync(stream);
-                stream = nullptr;
-
-                // launch file after creation
-                auto launched = co_await Windows::System::Launcher::LaunchFileAsync(file);
-                if (!launched)
-                {
-                    DisplayErrorMessage("File has been created:\n" + file->Path);
-                }
+                DisplayErrorMessage("File has been created:\n" + file->Path);
             }
         }
-        catch (Exception^ ex)
-        {
-            DisplayErrorMessage(ex->Message);
-        }
+        if (frameGrabber)
+            delete frameGrabber;
```

Comment:
I think this is unneccessary. The FrameGrabber is deleted as soon as it gets out of scope.